driver/src/price_finding/optimization_price_finder.rs

```diff
@@ -25,7 +28,10 @@ use std::time::Duration;
 /// the serialization cannot be controlled.
 #[derive(Copy, Clone, Debug, Default, Eq, PartialEq, Deserialize, Serialize)]
 #[serde(transparent)]
-pub struct Num(#[serde(with = "display_fromstr")] pub u128);
+pub struct Num<T>(#[serde(with = "display_fromstr")] pub T)
+where
+    T: Copy + Clone + Debug + Default + Eq + PartialEq + Display + FromStr,
```

Comment:
nit: built-in derive proc macros add appropriate trait bounds automatically:
```suggestion
    T: Display + FromStr,
```